airflow/models.py

```diff
@@ -3500,6 +3500,33 @@ def deactivate_stale_dags(expiration_date, session=None):
             session.merge(dag)
             session.commit()
 
+    @staticmethod
+    def get_num_task_instances(session, dag_id, task_ids, states=None):
```

Comment:
session as first argument? We don't do that anywhere. I think you can aso use @provide_session here btw so then you can move it to the end.